cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/namespace/StorageProviderNamespaceAdminTest.java

```diff
@@ -76,5 +84,31 @@ public void test() throws Exception {
     Assert.assertTrue(namespaceLocation.exists());
     storageProviderNamespaceAdmin.delete(myspace);
     Assert.assertFalse(namespaceLocation.exists());
+
+    // test for custom location mapped namespace
+    myspace = new NamespaceId("custom");
+    myspaceMeta = new NamespaceMeta.Builder().setName(myspace.getNamespace()).setRootDirectory("custom").build();
+    try {
+      storageProviderNamespaceAdmin.create(myspaceMeta);
+      Assert.fail("Expected exception to be thrown while creating namespace earlier");
+    } catch (Exception e) {
+      Assert.assertTrue(e instanceof IOException);
+    }
+    // the delete will look up meta so store that too
+    namespaceStore.create(myspaceMeta);
+    // now create the home directory too
+    // create the custom location
+    Location customLocation = locationFactory.create("custom");
+    customLocation.mkdirs();
```

Comment:
let's move the creation after the ``storageProviderNamespaceAdmin.create(myspaceMeta)`` statement. That way, after creating the namespace via SPNA, you can assert that the ``Location`` does not exist. Then create it manually, followed by delete namespace.